server/controllers/chefProfileController.js

```diff
@@ -0,0 +1,132 @@
+const ChefProfile = require("../models/chefProfileModel");
+const AsyncHandler = require("express-async-handler");
+
+const createChefProfile = async (req, res) => {
+  try {
+    console.log("create chef profile controller reached");
+
+    // get authenticated user from middleware
+    const { user } = req;
+
+    // if authenticated user is not chef
+    if (!user.isChef) throw "Only chef can create chef profile";
```

Comment:
It would be better to `throw new Error("error message")`

Reply:
Fixed all throw messages